src/datasources/tag/TagDataSource.ts

```diff
@@ -145,7 +157,20 @@ export class TagDataSource extends DataSourceBase<TagQuery, DataSourceJsonData>
     }
   }
 
-  private async getMostRecentTags(path: string, workspace: string) {
+  private generatePathsFromTemplate(query: TagQuery) {
+    let paths: string[] = [query.path];
+    if (this.templateSrv.containsTemplate(query.path)) {
+      const replacedPath = this.templateSrv.replace(
+        query.path,
+        undefined,
+        (v: string | string[]): string => `{${v}}`
```

Comment:
I think this would have been easier if we just fetched all the [variables on the dashboard](https://github.com/grafana/grafana/blob/e8f23bb2cf666c4b380fc6f9f4e94c0ba5b1b326/public/app/features/templating/template_srv.ts#L77) and did the replacement oursevles. Then we wouldn't have to do all the regex matching stuff with the brackets. Not going to block the review on it though.